worlds/raft/__init__.py

```diff
@@ -108,19 +110,33 @@ def create_items(self):
         if frequencyItems:
             self.place_frequencyItems(frequencyItems)
 
+    def get_filler_item_name(self) -> str:
+        # A normal Raft world will have an extraItemNamePool defined after create_items.
+        if self.extraItemNamePool:
+            return self.random.choice(self.extraItemNamePool)
+
+        # If this is a "fake" world, e.g. item links with link replacement: Resource packs are always be safe to create
+        minRPSpecified = self.options.minimum_resource_pack_amount.value
+        maxRPSpecified = self.options.maximum_resource_pack_amount.value
+        minimumResourcePackAmount = min(minRPSpecified, maxRPSpecified)
+        maximumResourcePackAmount = max(minRPSpecified, maxRPSpecified)
+        resource_amount = self.random.randint(minimumResourcePackAmount, maximumResourcePackAmount)
+        resource_type = self.random.choice(resourcePackItems)
+        return createResourcePackName(resource_amount, resource_type)
+
     def set_rules(self):
         set_rules(self.multiworld, self.player)
 
     def create_regions(self):
         create_regions(self.multiworld, self.player)
-    
-    def create_item_replaceAsNecessary(self, name: str) -> Item:
-        isFrequency = "Frequency" in name
-        shouldUseProgressive = bool((isFrequency and self.options.island_frequency_locations == self.options.island_frequency_locations.option_progressive)
-            or (not isFrequency and self.options.progressive_items))
-        if shouldUseProgressive and name in progressive_table:
-            name = progressive_table[name]
-        return self.create_item(name)
+
+    def replace_item_name_as_necessary(self, name: str) -> str:
+        if "Frequency" in name:
+            if self.options.island_frequency_locations == self.options.island_frequency_locations.option_progressive:
+                return progressive_table[name]
+        elif self.options.progressive_items and name in progressive_table:
+            return progressive_table[name]
+        return name
```

Comment:
Confirmed that this crashes if "Sweep net" is renamed to "Sweep Frequency" since it isn't in the table. There's like forty ways to fix this, but here's one:
```suggestion
    def replace_item_name_as_necessary(self, name: str) -> str:
        if name not in progressive_table:
            return name
        if "Frequency" in name:
            if self.options.island_frequency_locations == self.options.island_frequency_locations.option_progressive:
                return progressive_table[name]
        elif self.options.progressive_items:
            return progressive_table[name]
        return name
```

Reply:
It's true that it's not a 1:1 conversion, but is there anything actually wrong with the new code? 🤔 Like you had to invent an item that doesn't exist to crash this code
And I would actually say that this crash is "good", since it informed you that your check for whether something is a frequency is wrong now

Reply:
Then I'd say the check is wrong now, since it gives false positives

Reply:
I don't agree with this change but it's such a minute detail in comparison to how an unresolved change request would look that I'll just yield